architecture/libraries.rst

```diff
@@ -4,24 +4,88 @@
 Libraries
 #########
 
+.. _libutils:
+
+libutils
+*******
+
+OP-TEE core and OP-TEE development kit for Trusted Application provide a
+libc standard library that is named **libtuils**. It implements many
+standard functions like ``snprintf()``, ``strncmp()``, ``memcpy()``,
+``malloc()``. ``qsort()``, and many more but not all standard libc functions.
+
+Note however that Trusted Applications implemented in "C" should use GP TEE
+Internal Core API functions rather than their stdlib equivalent. This makes
+those TAs implementation more portable to other GP TEE compliant environments.
+
+When ``CFG_ULIBS_SHARED`` is enabled, **libutils** is assigned UUID
+**71855bba-6055-4293-a63f-b0963a737360**.
+
 .. _libutee:
 
 libutee
 *******
 The :ref:`tee_internal_core_api` describes services that are provided to Trusted
 Applications. **libutee** is a library that implements this API.
 
-libutee is a static library the Trusted Applications shall statically link
-against. Trusted Applications do execute in non-privileged secure userspace and
-libutee also aims at being executed in the non-privileged secure userspace.
+libutee is designed as a userland library specifically dedicated to OP-TEE
+Trusted Applications and aims at being executed in the non-privileged secure
+userspace.
 
 Some services for this API are fully statically implemented inside the libutee
 library while some services for the API are implemented inside the OP-TEE core
 (privileged level) and libutee calls such services through system calls.
 
-.. _libmpa:
+When ``CFG_ULIBS_SHARED`` is enabled, **libutee** is assigned UUID
+**4b3d937e-d57e-418b-8673-1c04f2420226**.
 
-libmpa
+libmbedtls
+**********
+
+OP-TEE OS source tree provides support of the mbed TLS library, named
+**libmbedtls**.
+
+A specific build sequence compiles an instance of **libmbedtls** and links
+it to OP-TEE. Another build sequence compiles an instance of **libmbedtls**
+that will be linked with Trusted Applications.
+
+When ``CFG_ULIBS_SHARED`` is enabled, **libmbedtls** is assigned UUID
+**87bb6ae8-4b1d-49fe-9986-2b966132c309**.
+
+libunw
 ******
-Now deprecated, used to provide the BigNum library in OP-TEE.
+
+OP-TEE OS source tree implements execution stack back trace debug facilities
+available to both OP-TEE core and Trusted Applications. The feature relies
```

Comment:
The core doesn't use it directly, ldelf does.

Reply:
I see some libunw functions called from the core. For example `print_stack_arm{32|64}()` functions are called from `print_kernel_stack()` at [core/arch/arm/kernel/abort.c:82](https://github.com/OP-TEE/optee_os/tree/4.0.0/core/arch/arm/kernel/abort.c#L82) and [core/arch/arm/kernel/abort.c:69](https://github.com/OP-TEE/optee_os/tree/4.0.0/core/arch/arm/kernel/abort.c#L69).

Reply:
Ah, right, my mistake. libunw is indeed built twice for core and ldelf (but not user space TAs).